instance/models/server.py

```diff
@@ -145,7 +150,8 @@ def _set_status(self, status):
 
         self.status = status
         self.logger.info('Changed status: %s', self.status)
-        self.save()
+        if commit:
+            self.save()
```

Comment:
Actually, it was not to save db writes. In the update_status method, there's a place where we are changing the status to ERROR but we can't save the instance right away, because we still need to set the error field accordingly. This change was to support that.